central/complianceoperator/v2/pruner/pruner.go

```diff
@@ -53,6 +68,31 @@ func (p *pruneImpl) RemoveComplianceResourcesByCluster(ctx context.Context, clus
 	if err := p.scanResultDS.DeleteResultsByCluster(ctx, clusterID); err != nil {
 		log.Errorf("failed to delete scan result for cluster %s: %v", clusterID, err)
 	}
+
+	if err := p.compRuleDS.DeleteRulesByCluster(ctx, clusterID); err != nil {
+		log.Errorf("failed to delete cluster from rule for cluster id  %s: %v", clusterID, err)
+	}
+
+	if err := p.profileDS.DeleteProfileForCluster(ctx, "", clusterID); err != nil {
```

Comment:
Does this work?  The intent of that method was to remove a specific profile on the cluster.  If this works we would need to make sure that it will consistently continue to work as search framework is updated and write a test case that would fail if for some reason a change was introduced such that passing the empty string didn't evaluate to everything.

and we would certainly need to write a comment detailing what passing the empty string here means.

Reply:
I have added a new function to delete profile of a cluster